mozjs-sys/build.rs

```diff
@@ -91,6 +90,34 @@ fn main() {
     }
 }
 
+/// Check env variable conditions to decide if we need to link pre-built archive first.
+/// And then return bool value to notify if we need to build from source instead.
+fn should_build_from_source(build_dir: &Path) -> bool {
+    if env::var_os("MOZJS_FROM_SOURCE").is_some() {
+        println!("Environment variable MOZJS_FROM_SOURCE is set. Building from source directly.");
+        true
+    } else if env::var_os("MOZJS_CREATE_ARCHIVE").is_some() {
+        println!(
+            "Environment variable MOZJS_CREATE_ARCHIVE is set. Building from source directly."
+        );
+        true
+    } else if env::var_os("CARGO_FEATURE_DEBUGMOZJS").is_some() {
+        println!("debug-mozjs feature is enabled. Building from source directly.");
+        true
+    } else if !env::var_os("CARGO_FEATURE_STREAMS").is_some() {
+        println!("streams feature isn't enabled. Building from source directly.");
+        true
+    } else {
+        match link_static_lib_binaries(&build_dir) {
+            Ok(()) => false,
+            Err(e) => {
+                println!("cargo:warning=Failed to link pre-built archive by {e}. Building from source instead.");
+                true
+            }
+        }
```

Comment:
Please move this of the helper and call it based on the return value. For instance:

```rust
let mut build_from_source = should_build_from_source(&build_dir);
if !build_from_source && link_static_lib_binaries(&build_dir).is_err() {
    println!("cargo:warning=Failed to link pre-built archive by {e}. Building from source instead.");
    build_from_source = true;
}
```